src/Honeybadger.php

```diff
@@ -109,13 +109,23 @@ public function checkin(string $key): void
     }
 
     /**
-     * @param  int|string  $key
-     * @param  int|string|array  $value
-     * @return void
+     * Accepts a key and value argument or alternatively a key-value array as sole argument.
+     *
+     * @param  int|string|array  $key
+     * @param  int|string|array|null  $value
+     * @return self
      */
-    public function context($key, $value): void
+    public function context($key, $value = null)
     {
-        $this->context->set($key, $value);
+        if (is_array($key)) {
```

Comment:
A user shouldn't be passing an array as a key, but I favour having the check for the number of arguments there. That way, it's clear what the expected sage is.

Reply:
I disagree with this, if we just check number of arguments we assume that a single argument must be an array, but it's still possible to pass a single string for example, which then would crash in the foreach.

A check for is_array is safer because then the code will act upon what was actually received.

Reply:
I meant a check for both, like in my example.

Reply:
I'm in favor of safely ignoring the second argument if the first is an array, if we check the number of arguments and is_array, then it's possible to end up in the situation where an array and a second argument (of any type) was passed, then the else case will fail.

The best option imo would be to throw an InvalidArgumentException if any other arguments are passed along with an array, what do you think of that? I think we get the best of both use cases then.

Reply:
Nah, I think this is fine.